include/seqan3/alignment/pairwise/detail/policy_alignment_result_builder.hpp

```diff
@@ -0,0 +1,111 @@
+    using traits_type = alignment_configuration_traits<alignment_configuration_t>;
+    //!\brief The alignment result type.
+    using result_type = typename traits_type::alignment_result_type;
+
+    static_assert(!std::same_as<result_type, empty_type>, "The alignment result type was not configured.");
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    policy_alignment_result_builder() = default; //!< Defaulted.
+    policy_alignment_result_builder(policy_alignment_result_builder const &) = default; //!< Defaulted.
+    policy_alignment_result_builder(policy_alignment_result_builder &&) = default; //!< Defaulted.
+    policy_alignment_result_builder & operator=(policy_alignment_result_builder const &) = default; //!< Defaulted.
+    policy_alignment_result_builder & operator=(policy_alignment_result_builder &&) = default; //!< Defaulted.
+    ~policy_alignment_result_builder() = default; //!< Defaulted.
+
+    /*!\brief Construction and initialisation using the alignment configuration.
+     * \param[in] config The alignment configuration [not used in this context].
+     */
+    policy_alignment_result_builder(alignment_configuration_t const & SEQAN3_DOXYGEN_ONLY(config))
+    {}
+    //!\}
+
+    /*!\brief Builds the seqan3::alignment_result based on the given alignment result type and then invokes the
+     *        given callable with the result.
+     *
+     * \tparam indexed_sequence_pair_t The type of the indexed sequence pair.
+     * \tparam score_t The type of the score.
+     * \tparam callback_t The type of the callack to invoke.
+     *
+     * \param[in] indexed_sequence_pair The indexed sequence pair [maybe unused].
+     * \param[in] score The best alignment score [maybe unused].
+     * \param[in] callback The callback to invoke with the generated result.
+     *
+     * \details
+     *
+     * Generates a seqan3::alignment_result object with the results computed during the alignment. Depending on the
+     * seqan3::align_cfg::result configuration only the requested values are stored. In some cases some additional
+     * work is done to generate the requested result. For example computing the associated alignment from the traceback
+     * matrix.
+     */
+    template <typename indexed_sequence_pair_t, typename score_t, typename callback_t>
+    //!\cond
+        requires std::invocable<callback_t, result_type>
+    //!\endcond
+    void build_result_and_send([[maybe_unused]] indexed_sequence_pair_t && indexed_sequence_pair,
+                               [[maybe_unused]] score_t score,
+                               callback_t && callback)
+    {
+        using std::get;
+        using invalid_t = std::nullopt_t *;
+
+        result_type result{};
+
+        static_assert(!std::same_as<decltype(result.data.id), invalid_t>,
+                      "Invalid configuration. Expected result with id!");
+        result.data.id = std::move(get<1>(indexed_sequence_pair));
```

Comment:
? `indexed_sequence_pair` is used?

Reply:
right, I was referring to the sequences and forgot that I extract the ID already.

Reply:
Ah no, I was referring to the future here already since we said we want to have the user decide what to get and what not. So it might happen that the sequences and id are not used.

Reply:
I would keep it then, ok?